libs/pilight/protocols/433.92/daycom.c

```diff
@@ -0,0 +1,251 @@
+                return EXIT_FAILURE;
+        } else if(id > 63 || id < 0) {
+                logprintf(LOG_ERR, "daycom: invalid id range");
+                return EXIT_FAILURE;
+        } else if(systemcode > 16383 || systemcode < 0) {
+                logprintf(LOG_ERR, "daycom: invalid systemcode range");
+                return EXIT_FAILURE;
+        } else if(unit > 7 || unit < 0) {
+                logprintf(LOG_ERR, "daycom: invalid unit range");
+                return EXIT_FAILURE;
+        } else {
+                daycomCreateMessage(id, systemcode, unit, state);
+                daycomClearCode();
+                daycomCreateId(id);
+                daycomCreateSystemCode(systemcode);
+                daycomCreateState(state);
+                daycomCreateUnit(unit);
+                daycomCreateFooter();
+		daycom->rawlen = RAW_LENGTH;
+		state = 0;
+        }
+        return EXIT_SUCCESS;
+}
+
+static void daycomPrintHelp(void) {
+	printf("\t -i --id=Ident\t the Transmitter\n");
+	printf("\t -s --systemcode=systemcode\tcontrol a device with this systemcode\n");
+	printf("\t -u --unitcode=unitcode\t\tcontrol a device with this unitcode\n");
+	printf("\t -t --on\t\t\tsend an on signal\n");
+	printf("\t -f --off\t\t\tsend an off signal\n");
+}
+
+#if !defined(MODULE) && !defined(_WIN32)
+__attribute__((weak))
+#endif
+
+void daycomInit(void) {
+
+	protocol_register(&daycom);
+	protocol_set_id(daycom, "daycom");
+	protocol_device_add(daycom, "daycom", "Daycom Switches");
+        daycom->devtype = SWITCH;
+        daycom->hwtype = RF433;
+        daycom->minrawlen = RAW_LENGTH;
+        daycom->maxrawlen = RAW_LENGTH;
+        daycom->maxgaplen = MAX_PULSE_LENGTH*PULSE_DIV;
+        daycom->mingaplen = MIN_PULSE_LENGTH*PULSE_DIV;
+
+        options_add(&daycom->options, 't', "on", OPTION_NO_VALUE, DEVICES_STATE, JSON_STRING, NULL, NULL);
+        options_add(&daycom->options, 'f', "off", OPTION_NO_VALUE, DEVICES_STATE, JSON_STRING, NULL, NULL);
+        options_add(&daycom->options, 'u', "unit", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, "^([0-7])$");
+        options_add(&daycom->options, 's', "systemcode", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, "^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$");
+        options_add(&daycom->options, 'i', "id", OPTION_HAS_VALUE, DEVICES_ID, JSON_NUMBER, NULL, "^[0-6]?[0-9]$");
+
+	options_add(&daycom->options, 0, "readonly", OPTION_HAS_VALUE, GUI_SETTING, JSON_NUMBER, (void *)0, "^[10]{1}$");
+
+        daycom->parseCode=&daycomParseCode;
+        daycom->createCode=&daycomCreateCode;
+        daycom->printHelp=&daycomPrintHelp;
+        daycom->validate=&validate;
+}
+#if defined(MODULE) && !defined(_WIN32)
+void compatibility(struct module_t *module) {
+        module->name = "daycom";
+        module->version = "0.02";
+        module->reqversion = "6.0";
+        module->reqcommit = "2";
```

Comment:
Actually check what is the needed commit nr. Check the other protocols as a reference.